Crash to desktop when attempting to open new tab

Just now.. Eight tabs open. Try to open new tab with "+" button on taps screen. Display drops back to my normal phone homescreen. Firefox doesn't seem to have crashed as it still appears in the running apps, but when I manually go back to FF the new tab has not opened.

Copied from About/Crashes...
c6f2f144-dd97-4587-a419-a4ac8edb3fae
java.lang.RuntimeException: android.os.DeadSystemException
----
java.lang.RuntimeException: android.os.DeadSystemException
	at android.content.ClipboardManager.getPrimaryClip(ClipboardManager.java:137)
	at org.mozilla.fenix.utils.ClipboardHandler.getText(ClipboardHandler.kt:2)
	at org.mozilla.fenix.utils.ClipboardHandler.extractURL(ClipboardHandler.kt:1)
	at org.mozilla.fenix.search.SearchDialogFragment$observeClipboardState$1$2.emit(SearchDialogFragment.kt:25)
	at mozilla.components.support.ktx.kotlinx.coroutines.flow.FlowKt$ifChanged$$inlined$filter$1$2.emit(Emitters.kt:9)
	at org.mozilla.fenix.search.SearchDialogFragment$observeClipboardState$1$invokeSuspend$$inlined$map$1$2.emit(Emitters.kt:14)
	at mozilla.components.lib.state.ext.FragmentKt$consumeFlow$1$invokeSuspend$$inlined$filter$1$2.emit(Emitters.kt:8)
	at kotlinx.coroutines.flow.FlowKt__ChannelsKt.emitAllImpl$FlowKt__ChannelsKt(Channels.kt:13)
	at kotlinx.coroutines.flow.FlowKt__ChannelsKt$emitAllImpl$1.invokeSuspend(Channels.kt:1)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:4)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:18)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:359)
	at android.app.ActivityThread.main(ActivityThread.java:7418)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:935)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@17f8647, Dispatchers.Main]
Caused by: android.os.DeadSystemException
	... 18 more

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21866)

--- a/app/src/main/java/org/mozilla/fenix/utils/ClipboardHandler.kt
+++ b/app/src/main/java/org/mozilla/fenix/utils/ClipboardHandler.kt
@@ -11,6 +11,7 @@ import android.os.Build
 import android.view.textclassifier.TextClassifier
 import androidx.annotation.VisibleForTesting
 import androidx.core.content.getSystemService
+import mozilla.components.support.ktx.kotlin.MAX_URI_LENGTH
 import mozilla.components.support.utils.SafeUrl
 import mozilla.components.support.utils.WebURLFinder
 import org.mozilla.fenix.perf.Performance.logger
@@ -58,6 +59,10 @@ class ClipboardHandler(val context: Context) {
      */
     fun extractURL(): String? {
         return text?.let {
+            if (it.length > MAX_URI_LENGTH) {
+                return null
+            }
+
             val finder = WebURLFinder(it)
             finder.bestWebURL()
         }

